packages/mds-ingest-service/repository/index.ts

```diff
@@ -473,21 +473,23 @@ class IngestReadWriteRepository extends ReadWriteRepository {
   public getLatestTelemetryForDevices = async (device_ids: UUID[]): Promise<TelemetryDomainModel[]> => {
     try {
       const connection = await this.connect('ro')
+      /* sets enable_bitmapscan = off for this connection the duration of the session */
+      await connection.getRepository(TelemetryEntity).query(`select set_config('enable_bitmapscan','off', false)`)
+
       const entities = await connection
         .createQueryBuilder(TelemetryEntity, 'telemetry')
-        .innerJoin(
-          subquery =>
-            subquery
-              .select('id, RANK() OVER (PARTITION BY device_id ORDER BY timestamp DESC) AS rownum')
-              .where('device_id = ANY(:device_ids)', { device_ids })
-              .from(TelemetryEntity, 't'),
-          'last_device_telemetry',
-          'last_device_telemetry.id = telemetry.id AND last_device_telemetry.rownum = 1'
-        )
+        .distinctOn(['device_id'])
+        .orderBy('device_id', 'DESC')
+        .addOrderBy('timestamp', 'DESC')
         .getMany()
       return entities.map(TelemetryEntityToDomain.mapper())
     } catch (error) {
       throw RepositoryError(error)
+    } finally {
+      const connection = await this.connect('ro')
+      /* sets enable_bitmapscan = on for this connection the duration of the session. Seems like it shouldn't be an issue,
+         but connection pooling could/should be in use, and persistent connections are used between queries. */
```

Comment:
Is there any risk of this connection getting used by something else before this fires?

Reply:
Maybe we put this into a transaction and set the last argument to `true`?
https://www.postgresql.org/docs/9.3/functions-admin.html
> set_config sets the parameter setting_name to new_value. If is_local is true, the new value will only apply to the current transaction. If you want the new value to apply for the current session, use false instead. The function corresponds to the SQL command SET.

Reply:
found an API for this https://github.com/typeorm/typeorm/blob/master/docs/transactions.md#using-queryrunner-to-create-and-control-state-of-single-database-connection ! going to update it

Reply:
worked great, and there's less code!